### PR TITLE
Add a sample for VK_EXT_external_memory_host

### PR DESCRIPTION
**Pull request by apazylbe**
Only VK_EXTERNAL_MEMORY_HANDLE_TYPE_HOST_ALLOCATION_BIT_EXT for now
VkBufferExternalMemoryHost is a VkBuffer that uses external
host allocated memory as backing memory for the buffer.

### CODE REVIEW

**Comment by apazylbe** [comment]
@AWoloszyn @aellem would you be able to review? I can't seem to edit reviewers

File: application_sandbox/external_memory_host/main.cpp [hunk 1]
```diff
@@ -0,0 +1,498 @@
...
+      allocation_size_ = vulkan::RoundUp(
+          allocation_size_, physical_device_external_memory_host_properties
+                                .minImportedHostPointerAlignment);
+    }
+
+#ifdef _WIN32
+    host_pointer_ = VirtualAlloc(0, allocation_size_, MEM_RESERVE | MEM_COMMIT,
+                                 PAGE_READWRITE);
+#elif __linux__
+    host_pointer_ = mmap(nullptr, allocation_size_, PROT_READ | PROT_WRITE,
```
Comment (AWoloszyn) [inline application_sandbox/external_memory_host/main.cpp:110]: Where to the flags for this come from (specificaly the MAP_* flags). The spec seems silent on these requirements.

File: application_sandbox/external_memory_host/main.cpp [hunk 2]
```diff
@@ -0,0 +1,498 @@
...
+                         MAP_PRIVATE | MAP_NORESERVE | MAP_ANONYMOUS, -1, 0);
+#endif
+    VkMemoryHostPointerPropertiesEXT host_pointer_properties = {
+        VK_STRUCTURE_TYPE_MEMORY_HOST_POINTER_PROPERTIES_EXT};
+    device_->vkGetMemoryHostPointerPropertiesEXT(
```
Comment (AWoloszyn) [inline application_sandbox/external_memory_host/main.cpp:115]: Although we are allocating extra memory for minImportedHostPointerAlignment, we never actually try and align host_pointer_ to the required alignment, which  vkGetMemoryHostPointerPropertiesEXT requires.

File: application_sandbox/external_memory_host/main.cpp (hunk 1 above)
Comment (apazylbe) [reply application_sandbox/external_memory_host/main.cpp]: Removed now, just using aligned alloc

File: application_sandbox/external_memory_host/main.cpp (hunk 2 above)
Comment (apazylbe) [reply application_sandbox/external_memory_host/main.cpp:126]: Done.

File: application_sandbox/external_memory_host/main.cpp [hunk 3]
```diff
@@ -0,0 +1,505 @@
...
+
+    if (physical_device_external_memory_host_properties
+            .minImportedHostPointerAlignment > 0) {
+      allocation_size_ = vulkan::RoundUp(
+          allocation_size_, physical_device_external_memory_host_properties
+                                .minImportedHostPointerAlignment);
+    }
+
+    host_pointer_ =
+        aligned_alloc(physical_device_external_memory_host_properties
```
Comment (AWoloszyn) [inline application_sandbox/external_memory_host/main.cpp:120]: can we use the C++ version of this (not 100% sure if msvc implements the C11 version)
std::aligned_alloc

Comment (apazylbe) [reply application_sandbox/external_memory_host/main.cpp:120]: That would require C++ 17, right? 
I don't think the project right now requires/specifies use of C++ 17, does it?
I can also use malloc and then align it manually.. if that sounds like a better option

Comment (apazylbe) [reply application_sandbox/external_memory_host/main.cpp:120]: @AWoloszyn any thoughts on this?

Comment (AWoloszyn) [reply application_sandbox/external_memory_host/main.cpp:120]: If this builds on Windows,  then I am happy with it.

**Review by AWoloszyn — APPROVED** [review]

**Pull request merged**